- Buscamos maximizar su retorno de inversión (ROI)

 Se calcula como la ganancia o pérdida en relación con el costo de la inversión original, y depende de varios factores, como el precio de compra y venta de las acciones, las comisiones y los impuestos

In [2]:
#Librerias necesarias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import yfinance as yf
import plotly.graph_objs as go
from sklearn.linear_model import LinearRegression
import datetime as dt

In [3]:
# Descargar datos históricos para rendimiento total
tickers_sec = ["XLY", "XLK", "XLV", "XLI"]
data_sect_total = yf.download(tickers_sec, start="2000-01-01", end="2023-02-23")

[*********************100%***********************]  4 of 4 completed


In [4]:
# Descargar datos históricos para rendimiento total
tickers_seclec = [ "XLY", "XLP", "XLE", "XLF", "XLV", "XLI", "XLK", "XLU", "XLB"]
data_sect_rec = yf.download(tickers_sec, start="2000-01-01", end="2023-02-23")

[*********************100%***********************]  4 of 4 completed


In [5]:
#Vistazo
data_sect_rec.head()

Adj Close                                       Close            \
                  XLI        XLK        XLV        XLY        XLI       XLK   
Date                                                                          
2000-01-03  18.979511  42.209305  21.571774  22.924370  28.843750  55.43750   
2000-01-04  18.455160  40.067902  21.082270  22.234735  28.046875  52.62500   
2000-01-05  18.372904  39.473095  20.893141  21.961271  27.921875  51.84375   
2000-01-06  18.619658  38.164444  20.959888  22.210953  28.296875  50.12500   
2000-01-07  19.329077  38.830658  21.215769  23.257301  29.375000  51.00000   

                                       High            ...        Low  \
                  XLV        XLY        XLI       XLK  ...        XLV   
Date                                                   ...              
2000-01-03  30.296875  30.125000  29.484375  55.62500  ...  30.062500   
2000-01-04  29.609375  29.218750  28.609375  54.75000  ...  29.531250   
2000-01-05  29.343750  28.859375  28.125000  53.06250  ...  29.218750   
2000-01-06  29.437500  29.187500  28.375000  52.09375  ...  29.156250   
2000-01-07  29.796875  30.562500  29.375000  51.00000  ...  29.359375   

                            Open                               Volume  \
                  XLY        XLI     XLK       XLV        XLY     XLI   
Date                                                                    
2000-01-03  29.859375  29.484375  55.625  31.00000  31.000000   82800   
2000-01-04  29.031250  28.468750  54.000  30.18750  29.687500  146900   
2000-01-05  28.718750  28.000000  51.750  29.78125  29.343750  129200   
2000-01-06  29.000000  27.921875  51.250  29.31250  29.046875   54000   
2000-01-07  29.500000  28.625000  49.625  29.46875  29.500000   32900   

                                     
                XLK     XLV     XLY  
Date                                 
2000-01-03   938100  246700  212900  
2000-01-04   545500   35300  177200  
2000-01-05  1906000  137600  124600  
2000-01-06  1670000   50000  135800  
2000-01-07   686000   22100   70000  

[5 rows x 24 columns]

In [6]:
def calcular_roi_sectores(inversion_inicial):
    tickers_sec = ["XLY", "XLV", "XLI", "XLK"]
    data_sect_total = yf.download(tickers_sec, start="2000-01-01", end="2023-02-23")

    for sector in tickers_sec:
        # Obtener el precio de cierre ajustado más reciente
        precio_actual = data_sect_total["Adj Close"].tail(1)[sector]
        # Calcular el precio de cierre ajustado hace 1 año
        precio_hace_un_anio = data_sect_total["Adj Close"].shift(252).dropna()[sector][0]
        # Calcular el ROI para el último año
        ganancia = precio_actual - precio_hace_un_anio
        roi = (ganancia / precio_hace_un_anio) * 100
        # Calcular el valor de la inversión después de 1 año
        valor_final = inversion_inicial * (1 + roi / 100)
        # Imprimir los resultados para cada sector
        print(f"Para el sector {sector}:")
        print(f"Inversión inicial: ${inversion_inicial}")
        print(f"ROI para el último año: {round(roi, 2)}%")
        print(f"Valor final después de 1 año: ${round(valor_final, 2)}\n")

In [8]:
calcular_roi_sectores(1000)

[*********************100%***********************]  4 of 4 completed
Para el sector XLY:
Inversión inicial: $1000
ROI para el último año: Date
2023-02-22    538.1
Name: XLY, dtype: float64%
Valor final después de 1 año: $Date
2023-02-22    6380.98
Name: XLY, dtype: float64

Para el sector XLV:
Inversión inicial: $1000
ROI para el último año: Date
2023-02-22    501.9
Name: XLV, dtype: float64%
Valor final después de 1 año: $Date
2023-02-22    6018.98
Name: XLV, dtype: float64

Para el sector XLI:
Inversión inicial: $1000
ROI para el último año: Date
2023-02-22    429.1
Name: XLI, dtype: float64%
Valor final después de 1 año: $Date
2023-02-22    5290.97
Name: XLI, dtype: float64

Para el sector XLK:
Inversión inicial: $1000
ROI para el último año: Date
2023-02-22    222.91
Name: XLK, dtype: float64%
Valor final después de 1 año: $Date
2023-02-22    3229.15
Name: XLK, dtype: float64



### Conclusión de ROI con una inversión inicial de U$D 1000:

- XLY(Consumo discrecional): 

El ROI para el sector en el ultimo año: 538.1%

El valor final de la inversión después de 1 año: U$D 6380.98

- XLV(Cuidado de la salud): 

El ROI para el sector en el ultimo año: 501.9%

El valor final de la inversión después de 1 año: U$D 6018.98

- XLI(Industriales): 

El ROI para el sector en el ultimo año:  429.1%

El valor final de la inversión después de 1 año: U$D 5290.97 

- XLK(Tecnología de la información): 

El ROI para el sector en el ultimo año: 222.9%  

El valor final de la inversión después de 1 año: U$D 3229.15